packages/nimble-components/src/select/styles.ts

```diff
@@ -37,8 +37,15 @@ export const styles = css`
         order: 1;
     }
 
-    [part='indicator'] {
+    .clear-button {
```

Comment:
Right now I can still click the clear button when the component is disabled. Can you give it disabled styling and make it non-interactive? 

Not sure if we also need to ensure ESC does nothing in that case; I don't see a way to tab focus a disabled control anyway. But we should follow whatever patterns we already use for other keyboard interactions when the select is disabled.

Reply:
It will now no longer be in the DOM when the `Select` is disabled.